jerry-core/ecma/base/ecma-helpers-number.c

```diff
@@ -370,7 +373,11 @@ ecma_number_remainder (ecma_number_t left_num, /**< left operand */
 
   if (ecma_number_is_zero (r) && ecma_number_is_negative (left_num))
   {
+#ifdef __APPLE__
+    r = ecma_number_negate (r);
+#else /* !__APPLE__ */
```

Comment:
Is this the right macro guard to use? If this is a compiler bug then is it OK to use a guard on the platform? Which versions of which compiler are affected? Is there an issue report for this anywhere that could be linked for reference? Is there no cli option for the compiler to ensure that signed/negative zeros are properly handled, to work around the issue without code base change?

Reply:
This bug only occurs on `macos-14`, which by default uses `Clang`, Version: `15.0.0.15000040`
this seems to be describing the same problem: https://github.com/llvm/llvm-project/issues/82744 (for the first issue, I was unable to find a report for the second)

These issues first popped up when `macos-latest` in github actions switched from `macos-13` to `macos-14`.
The easiest fix for now without a codebase change would be to go back to using `macos-13` for the time being, and switch back to `macos-latest` at a leter date when these issues have hopefully been resolved.